proxy/src/main/java/com/wavefront/agent/listeners/ZipkinPortUnificationHandler.java

```diff
@@ -0,0 +1,260 @@
+package com.wavefront.agent.listeners;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.RateLimiter;
+import com.wavefront.agent.auth.TokenAuthenticatorBuilder;
+import com.wavefront.agent.auth.TokenValidationMethod;
+import com.wavefront.agent.handlers.HandlerKey;
+import com.wavefront.agent.handlers.ReportableEntityHandler;
+import com.wavefront.agent.handlers.ReportableEntityHandlerFactory;
+import com.wavefront.data.ReportableEntityType;
+import com.yammer.metrics.Metrics;
+import com.yammer.metrics.core.Counter;
+import com.yammer.metrics.core.MetricName;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import wavefront.report.Annotation;
+import wavefront.report.Span;
+import zipkin2.SpanBytesDecoderDetector;
+import zipkin2.codec.BytesDecoder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Handler that processes Zipkin trace data over HTTP and converts them to Wavefront format.
+ *
+ * @author Anil Kodali (akodali@vmware.com)
+ */
+public class ZipkinPortUnificationHandler extends PortUnificationHandler {
+  private static final Logger logger = Logger.getLogger(
+          ZipkinPortUnificationHandler.class.getCanonicalName());
+  private final String handle;
+  private final ReportableEntityHandler<Span> handler;
+  private final AtomicBoolean traceDisabled;
+  private final RateLimiter warningLoggerRateLimiter = RateLimiter.create(0.2);
+  private final Counter discardedTraces;
+  private final Counter discardedBatches;
+  private final Counter processedBatches;
+  private final Counter failedBatches;
+
+  private final static Set<String> ZIPKIN_VALID_PATHS = ImmutableSet.of("/api/v1/spans/", "/api/v2/spans/");
+  private final static String ZIPKIN_VALID_HTTP_METHOD = "POST";
+  private final static String APPLICATION_KEY = "application";
```

Comment:
Can you share this across Jaeger and Zipkin? Ideally we should get this from tier 1 SDK
since are adding that dependency already, please use
https://github.com/wavefrontHQ/wavefront-sdk-java/blob/master/src/main/java/com/wavefront/sdk/common/Constants.java#L61